src/executor/graph_executor.cc

```diff
@@ -558,20 +558,31 @@ void GraphExecutor::InitArguments(const nnvm::IndexedGraph& idx,
                                   const std::unordered_set<std::string>& shared_arg_names,
                                   const Executor* shared_exec,
                                   std::unordered_map<std::string, NDArray>* shared_buffer,
+                                  std::vector<std::string>* in_names,
                                   std::vector<NDArray>* in_arg_vec,
                                   std::vector<NDArray>* arg_grad_vec,
                                   std::vector<NDArray>* aux_state_vec) {
+  // build name/index map for graph input nodes
+  std::map<std::string, uint32_t> inv_name_map;
```

Comment:
do you need ordering? could be unordered_map as is amortized constant vs log N

Reply:
Is it possible to map the source node instead of the argument name for efficiency and to avoid the problem @ptrendx mentions below?  I don't know if it's possible but looking up ptr is much faster than str.